BlockSettleUILib/Trading/DealerXBTSettlementContainer.cpp

```diff
@@ -426,18 +426,13 @@ void DealerXBTSettlementContainer::onSignedPayinRequested(const std::string& set
       return;
    }
 
-   if (usedPayinHash_.empty()) {
-      logger_->warn("[DealerXBTSettlementContainer::onSignedPayinRequested] "
-         "previously saved payin hash was empty");
-      usedPayinHash_ = payinHash;
-   }
-   if (payinHash.empty() || (usedPayinHash_ != payinHash)) {
-      SPDLOG_LOGGER_ERROR(logger_, "payin hash mismatch: {} vs {}"
-         , usedPayinHash_.toHexStr(true), payinHash.toHexStr(true));
-      failWithErrorText(tr("pay-in hash mismatch"), bs::error::ErrorCode::TxInvalidRequest);
+   if (payinHash.empty()) {
+      failWithErrorText(tr("Invalid Sign Pay-In request"), bs::error::ErrorCode::InternalError);
       return;
    }
 
+   expectedPayinHash_ = payinHash;
```

Comment:
why is this set twice?

Reply:
Different methods. First one is for sign payout request. And we need to send payin hash used to create payout. Second for signed payin, which we use to compare against signed TX hash